core-plugins/src/main/java/io/cdap/plugin/batch/source/XMLRecordReader.java

```diff
@@ -106,6 +107,8 @@ public void initialize(InputSplit split, TaskAttemptContext context) throws IOEx
     Configuration conf = context.getConfiguration();
     fs = file.getFileSystem(conf);
     XMLInputFactory factory = XMLInputFactory.newInstance();
+    enableExternalEntities = Boolean.valueOf(conf.get(XMLInputFormat.XML_INPUTFORMAT_ENABLE_EXTERNAL_ENTITIES));
```

Comment:
Can we also add supportDtd as a configuration option? I know you responded to my comment earlier about DTD support in XMLParser and XMLMultiParser, but from what I understand XMLInputFactory used in XMLRecordReader is a separate XML parser (StAX) from XMLParser and XMLMultiParser's DocumentBuilder (JAXP) and thus we will need a separate switch for both external entities and supporting DTD:
```
factory.setProperty(XMLInputFactory.SUPPORT_DTD, false);
```
[More Info here](https://cheatsheetseries.owasp.org/cheatsheets/XML_External_Entity_Prevention_Cheat_Sheet.html#xmlinputfactory-a-stax-parser).